bee-network/bee-autopeering/src/request.rs

```diff
@@ -0,0 +1,202 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    discovery::messages::{DiscoveryRequest, VerificationRequest},
+    hash,
+    local::Local,
+    packet::{msg_hash, MessageType},
+    peer::peer_id::PeerId,
+    peering::messages::PeeringRequest,
+    task::Repeat,
+    time::{self, Timestamp},
+};
+
+use tokio::sync::oneshot;
+
+pub(crate) use oneshot::channel as response_chan;
+
+use std::{
+    any::TypeId,
+    collections::HashMap,
+    fmt::Debug,
+    net::{IpAddr, SocketAddr},
+    sync::{Arc, RwLock, RwLockReadGuard, RwLockWriteGuard},
+    time::Duration,
+};
+
+type RequestHash = [u8; hash::SHA256_LEN];
+pub(crate) type ResponseTx = oneshot::Sender<Vec<u8>>;
+
+// If the request is not answered within that time it gets removed from the manager, and any response
+// coming in later will be deemed invalid.
+pub(crate) const REQUEST_EXPIRATION_SECS: u64 = 20;
```

Comment:
why not `Duration` instead?

Reply:
why asking such logical question? :trollface: 

Applied.